File: ForumAppCore/ForumAppCore/Entities/Question.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace ForumAppCore.Entities
{
    public class Question
    {
        [Key]
        public Guid Id { get; set; }
        public string Tittle { get; set; }
        public DateTime DatePosted { get; set; } = DateTime.Now;
        public long ViewCount { get; set; }
        public string NoiDung { get; set; }
        public AppUser AppUser { get; set; }
        public Guid UserId { get; set; }
        public ICollection<QuestionTag> QuestionTags { get; set; }
        public ICollection<CommentParent> CommentParents { get; set; }//Question - comment - cascade
    }
}

File: Client/src/app/_models/question.ts
import { CommentParent } from "./CommentParent";
import { QuestionTag } from "./questiontag";

export interface Question{
    id: string;
    tittle: string;
    datePosted: Date;
    viewCount: bigint;
    noiDung: string;
    userName: string;
    displayName: string;
    photoUrl: string;
    questionTags: QuestionTag[];
    commentParents: CommentParent[];
}

export interface QuestionAtHome{
    id: string;
    tittle: string;
    datePosted: Date;
    viewCount: bigint;
    noiDung: string;
    userName: string;
    displayName: string;
    photoUrl: string;
    questionTags: QuestionTag[];
    countCommentParents: number;
}

export interface SearchQuestion{
    tittle: string;
}